keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,819 @@
+// If rawSelector is present and requirements are empty, the rawSelector will be parsed and limited if the parsing succeeds.
+// If rawSelector is empty and requirements are present, the requirements should be honored
+// If rawSelector is present and requirements are present, the request is invalid.
+type FieldSelectorAttributes struct {
+	// rawSelector is the serialization of a field selector that would be included in a List query parameter.
+	// Webhook implementations must handle malformed rawSelectors.
+	RawSelector string
+
+	// requirements is the parsed interpretation of the rawSelector.
+	// All requirements must be met for a resource instance to match the selector.
+	// The two should match, but recall that clients can craft any kind of request.
+	// Webhook implementations must handle mismatched rawSelector and requirements, but how to handle them is up to the webhook.
+	Requirements []FieldSelectorRequirement
+}
+
+// LabelSelectorAttributes indicates a field limited access.
+// For webhooks:
+// The kube-apiserver will never send a request with rawSelector set, but we cannot control what other clients directly send.
+// If rawSelector is empty and requirements are empty, the request is not limited.
+// If rawSelector is present and requirements are empty, the request is not limited.
+// If rawSelector is empty and requirements are present, the requirements should be honored
+// If rawSelector is present and requirements are present, the request is invalid.
+// For the kube-apiserver:
+// If rawSelector is empty and requirements are empty, the request is not limited.
+// If rawSelector is present and requirements are empty, the rawSelector will be parsed and limited if the parsing succeeds.
+// If rawSelector is empty and requirements are present, the requirements should be honored
+// If rawSelector is present and requirements are present, the request is invalid.
+type LabelSelectorAttributes struct {
+	// rawSelector is the serialization of a field selector that would be included in a List query parameter.
+	// Webhook implementations must handle malformed rawSelectors.
+	RawSelector string
+
+    // requirements is the parsed interpretation of the rawSelector.
+    // All requirements must be met for a resource instance to match the selector.
+    // The two should match, but recall that clients can craft any kind of request.
+    // Webhook implementations must handle mismatched rawSelector and requirements, but how to handle them is up to the webhook.
+	Requirements []metav1.LabelSelectorRequirement
+}
+
+type FieldSelectorRequirement struct {
+	// key is the label key that the selector applies to.
+	Key string `json:"key" protobuf:"bytes,1,opt,name=key"`
+	// operator represents a key's relationship to a set of values.
+	// Valid operators are In, NotIn, Exists, DoesNotExist
+	Operator LabelSelectorOperator `json:"operator" protobuf:"bytes,2,opt,name=operator,casttype=LabelSelectorOperator"`
```

Comment:
since this is essentially an enum field, should describe how clients should handle unrecognized operators if we ever want the ability to expand this in the future

easiest way is to ignore entries with unrecognized operators and assume they don't limit the request